handlers.go

```
@@ -63,11 +62,61 @@ func (dh *DeviceHandlers) GetDeviceHandler(w http.ResponseWriter, r *http.Reques
 
 	dh.writeObject(w, device)
 
-	return
 }
 
-func (dh *DeviceHandlers) writeObject(w http.ResponseWriter, device *Device) {
-	respBody, err := json.Marshal(device)
+func (dh *DeviceHandlers) GetAllDevicesHandler(w http.ResponseWriter, r *http.Request) {
+	var ok bool
+	var limit, page int
+	keys := r.URL.Query()
+
+	limitStr := keys.Get("limit")
+	if limitStr == "" {
+		limit = 100
+	} else {
+		limit, ok = dh.stringIsPositiveNumberReturnInt(w, limitStr)
+		if !ok {
+			return
```

Why return? what will happen when I set `limit == a`? I see to possible solution:
- I can get 400 with message "invalid limit parameter" ;)
- the handler/service goes further with default value set

---

I have just looked at the implementation of `stringIsPositiveNumberReturnInt` - this will actually work but is not so obvious. The handler should return the errors (unless the method name suggests that it will do something in the name of handlerm, like `returnWhenError(w, err)` or something like that..)